code/books4u/api.py

```diff
@@ -16,3 +17,143 @@ def get_all_books(request):
         })
 
     return HttpResponse(json.dumps(response_data), content_type="application/json")
+
+def create_account(request):
+    if request.method == "POST":
+        try:
+            account_name = request.POST.get('name')
+            account_password = make_password(request.POST.get('password'))
+            account_e_mail = request.POST.get('e_mail')
+            account_gender = request.POST.get('gender')
+        except ValueError:
+            HttpResponse(json.dumps({'status': 'failure_get'}))
+        try:
+            new_user = User(name=account_name, password=account_password, e_mail=account_e_mail, gender=account_gender)
+            new_user.save()
+        except(TypeError, ValueError):
+            HttpResponse(json.dumps({'status': 'failure_save'}))
+        return HttpResponse(json.dumps({'status': 'success'}))
+    else:
+        return HttpResponse(json.dumps({'status': 'failure_method'}))
+
+""" used by users to add a new book
+    visibility of the new book is false, representing not officailly added
+"""
+def add_book(request):
+
+    if request.method == "POST":
+        try:
+            ISBN_ = request.POST.get('ISBN')
+            name_ = request.POST.get('name')
+            publish_date_ = request.POST.get('publish_date','')
+            edition_ = request.POST.get('edition','')
+            category_ = request.POST.get('category')
+            author_ = request.POST.get('author')
+        except requests.exceptions.RequestException:
```

Comment:
- requests.exceptions.RequestException is from python's requests module, I saw an example here https://stackoverflow.com/questions/16511337/correct-way-to-try-except-using-python-requests-module . I thought it would check for blank input, not sure. I will check the None manually anyway.
- I thought empty input for edition and publish firm were allowed, so I tried to deal with the case by giving default value = empty str, but it is easier to ask for all fields to be defined so I will remove them.

Reply:
We are not using python's requests module, the input 'request' is of django.http.HttpRequest.
Here is a [doc page](https://docs.djangoproject.com/en/2.0/ref/request-response/#module-django.http)